exercises/transpose/src/example/java/Transpose.java

```diff
@@ -0,0 +1,41 @@
+public class Transpose {
+    public String transpose(String toTranspose) {
+
+        StringBuilder transposedForm = new StringBuilder();
+        String[] rows = toTranspose.split("\n");
+
+        int maxRowSize = 0;
+
+        for (int i = 0; i < rows.length; i++) {
+            if (rows[i].length() > maxRowSize) {
+                maxRowSize = rows[i].length();
+            }
+        }
+
+        padRows(rows);
+
+        for (int newRowOldCol = 0; newRowOldCol < maxRowSize; newRowOldCol++) {
+            for (int newColOldRow = 0; newColOldRow < rows.length; newColOldRow++) {
+                if (newRowOldCol < rows[newColOldRow].length()) {
+                    transposedForm.append(rows[newColOldRow].charAt(newRowOldCol));
+                }
+            }
+
+            if (newRowOldCol != maxRowSize - 1) {
+                transposedForm.append("\n");
+            }
+        }
+
+        return transposedForm.toString();
+    }
+
+    private void padRows(String[] rows) {
+        for (int i = 0; i < rows.length; i++) {
+            for (int j = i; j < rows.length; j++) {
```

Comment:
No, and this is a bit tricky to see. I struggled with seeing this while making the solution. The problem description says the following:

>Pad to the left with spaces.
>Don't pad to the right.

You should only add padding in the case where there is a row to the right that is larger than it. Using the `maxRowSize` as you suggested works fine on all test cases except the last one, where the largest row is in the middle, but the last row is smaller, this causes the problem of extra spaces being inserted at which point you're technically padding to the right.